### PR TITLE
E_CLASSROOM-329 [FE/BE] Category Delete Function

### PR DESCRIPTION
**Pull request by ErickDoroy**
### Issue Link
- https://framgiaph.backlog.com/view/E_CLASSROOM-329
### Definition of Done
- [x] User can delete one category/subcategory
- [x] Confirmation modal should pop-up before deleting
- [x] A modal preventing the delete will pop-up if the selected category/subcategory has child subcategory

## Related PR:
- https://github.com/framgia/sph-classroom-els-be/pull/83
#### Main note
- http://localhost:3003/admin/categories?page=&search=&sortBy=&sortDirection=
- http://localhost:3003/admin/category-hierarchy
#### Pages Affected
- Admin Category page
- Admin Hierarchy page
- Admin page
- Quiz page

### Scenarios/ Test Cases
- [x] it should delete a category that don't have sub-category
- [x] it should show a modal that can't delete if their is a sub-category
#### User Interface
- [ ] when click the delete button confirmation will view
#### User Experience 
- N/A
#### Functionality
- N/A
### Screenshots
deletecategory
deletecategory1

### CODE REVIEW

**Comment by catampongan** [comment]
## Working
- [x] Displays confirmation modal when deleting a category
- [x] Able to delete a category/subcategory

## Comment
- [x] Can delete a category that has subcategory after deleting a category
- Steps to reproduce
 1. Delete a category that has no category
 2. Delete a category that has a category (Without refreshing the page}
- [x] Remove `cancel` button in the modal that can't delete a category
----------------------------------
task-329
task-329-2
task-329

-------------------------
- Total Test Time: 0.5 hr

**Comment by catampongan** [comment]
## Approved PR #163 
- [x] Displays confirmation modal when deleting a category
- [x] Able to delete a category/subcategory
- [x] Could not delete category that has subcategories
----------------------------------------------------
task-329
task-329
task-329-2

----------------------------
- Total Test Time: 0.75 hr
- Total Iterations: 1
- Total Comments: 2

File: src/components/ConfirmationModal/index.js [hunk 1]
```diff
@@ -67,7 +79,9 @@ ConfirmationModal.propTypes = {
...
   setShowModal: PropTypes.func,
   itemToDelete: PropTypes.string,
-  setDeleteConfirmed: PropTypes.func
+  setDeleteConfirmed: PropTypes.func,
+  message: PropTypes.string,
+  canDelete: PropTypes.bool
 };
 
-export default ConfirmationModal;
+export default ConfirmationModal;
```
Comment (karlolee50) [inline src/components/ConfirmationModal/index.js]: add extra line

File: src/pages/admin/categories/CategoryHierarchy/index.js [hunk 2]
```diff
@@ -160,7 +191,18 @@ const CategoryHierarchy = () => {
...
+                      className={style.kebabMenuItems}
+                      onClick={() => {
+                        if (category.subcategories_count <= 0) {
+                          setCanDelete(true);
+                        } else {
+                          setCanDelete(false);
+                        }
+                        setItemToDelete(category);
+                        setShowConfirmationModal(true);
+                      }}
```
Comment (karlolee50) [inline src/pages/admin/categories/CategoryHierarchy/index.js]: move to a function

File: src/pages/admin/categories/CategoryList/index.js [hunk 3]
```diff
@@ -79,7 +104,20 @@ const CategoryList = () => {
...
+                outline={true}
+                onClick={() => {
+                  if (category.subcategories_count <= 0) {
+                    setCanDelete(true);
+                  } else {
+                    setCanDelete(false);
+                  }
+                  setItemToDelete(category);
+                  setShowConfirmationModal(true);
+                }}
```
Comment (karlolee50) [inline src/pages/admin/categories/CategoryList/index.js]: move to a function. it looks similar to my previous comment. maybe you can use the same function?

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 10 minutes
Comment/s: 3

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**